c10/util/TypeCast.h

```diff
@@ -151,6 +152,17 @@ struct static_cast_with_inter_type<
   }
 };
 
+template <>
+struct static_cast_with_inter_type<
+    c10::complex<c10::Half>,
+    c10::Float8_e8m0fnu> {
+  C10_HOST_DEVICE __ubsan_ignore_undefined__ static inline c10::complex<
+      c10::Half>
+  apply(c10::Float8_e8m0fnu src) {
```

Comment:
curious what this is for?

Reply:
my non-satisfying reason is defining this cast is required for PyTorch to build.  There might be a way to refactor things to avoid that, but that's a bit outside of my expertise / what I think is important.

Reply:
This is needed because in Copy.cu we instantiate `gpu_kernel` that's supposed to handle any type to any type conversion. If instead we limit fp8 conversions to specific target dtype that we instantiate with `gpu_kernel_no_cast` this generic thing won't be needed, and tbh I don't see a point in converting any of fp8 types to anything but float or bfloat16 if we are generous.

Reply:
tracking in https://github.com/pytorch/pytorch/issues/146647